File: Go/1. Two Sum.go
package main

//goland:noinspection GoUnusedFunction
func twoSum(nums []int, target int) []int {
	numsMap := make(map[int]int)

	for i, num := range nums {
		if j, ok := numsMap[target-num]; ok {
			return []int{j, i}
		}
		numsMap[num] = i
	}

	return []int{}
}

File: Python/1. Two Sum.py
from typing import List


# noinspection PyMethodMayBeStatic, PyPep8Naming
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}

        for i, num in enumerate(nums):
            if target - num in map:
                return [map[target - num], i]
            map[num] = i

        return [-1, -1]
